src/CommonLib/Processors/ComputerAvailability.cs

```diff
@@ -89,23 +91,24 @@ public Task<ComputerStatus> IsComputerAvailable(ResolvedSearchResult result, ISe
             if (!_skipPasswordCheck)
             {
                 var passwordLastSet = Helpers.ConvertLdapTimeToLong(pwdLastSet);
+                var lastLogonTimeStamp = Helpers.ConvertLdapTimeToLong(lastLogon);
                 var threshold = DateTime.Now.AddDays(_computerExpiryDays * -1).ToFileTimeUtc();
 
-                if (passwordLastSet < threshold)
+                if (passwordLastSet < threshold && lastLogonTimeStamp < threshold)
```

Comment:
All of the `passwordLastSet` logic may be better served and documented by extracting it into a method.